packages/addons/src/plugins/localStoragePlugin.ts

```diff
@@ -21,44 +34,119 @@ export interface LocalStorageOptions {
+    // registered of FormKit type 'group' (eg. 'multi-step').
+    if (node.type !== 'group') return
+
+    const shouldUseLocalStorage = (controlNode: FormKitNode | undefined) => {
+      let controlFieldValue = true
+      if (controlNode) {
+        controlFieldValue = controlNode.value === true
+      }
+      return undefine(node.props.useLocalStorage) && controlFieldValue
+    }
 
-    node.on('created', () => {
-      const useLocalStorage = undefine(node.props.useLocalStorage)
-      if (!useLocalStorage) return
+    node.on('created', async () => {
+      node.addProps(['useLocalStorage'])
+      await node.settled
+
+      // if the user provided a control field, then we need to listen for changes
+      // and use it to determine whether or not to use local storage
+      const controlField = localStorageOptions?.control ?? undefined
+      let controlNode: FormKitNode | undefined
+      if (typeof controlField === 'string') {
+        const controlNode = node.at(controlField)
+        if (controlNode) {
+          controlNode.on('commit', () => {
+            useLocalStorage = shouldUseLocalStorage(controlNode)
+            if (!useLocalStorage) {
+              localStorage.removeItem(storageKey)
+            }
+          })
+        }
+      }
 
-      const prefix = LocalStorageOptions?.prefix ?? 'formkit'
-      const maxAge = LocalStorageOptions?.maxAge ?? 3600000 // 1 hour
-      const key = `${prefix}-${node.name}`
-      const value = localStorage.getItem(key)
+      let useLocalStorage = shouldUseLocalStorage(controlNode)
+      let saveTimeout: ReturnType<typeof setTimeout> | number = 0
+      const debounce =
+        typeof localStorageOptions?.debounce === 'number'
+          ? localStorageOptions.debounce
+          : 200
+      const prefix = localStorageOptions?.prefix ?? 'formkit'
+      const maxAge = localStorageOptions?.maxAge ?? 3600000 // 1 hour
+      const key = localStorageOptions?.key ? `-${localStorageOptions.key}` : '' // for scoping to a specific user
+      const storageKey = `${prefix}${key}-${node.name}`
 
-      if (value) {
-        const localStorageValue = JSON.parse(value)
-        if (localStorageValue.maxAge > Date.now()) {
-          node.input(localStorageValue.data)
+      const loadValue = async () => {
+        const value = localStorage.getItem(storageKey)
+        if (!value) return
+        const loadValue = JSON.parse(value)
+        if (typeof localStorageOptions?.beforeLoad === 'function') {
+          node.props.disabled = true
+          try {
+            loadValue.data = await localStorageOptions.beforeLoad(
+              loadValue.data
+            )
+          } catch (error) {
+            console.error(error)
+          }
+          node.props.disabled = false
+        }
+        if (!loadValue || typeof loadValue.data !== 'object') return
+        if (loadValue.maxAge > Date.now()) {
+          node.input(loadValue.data)
```

Comment:
It might be worth making this node.input synchronous.

```suggestion
          node.input(loadValue.data, false)
```